test/integration/physics_system.cc

```diff
@@ -2904,3 +2907,103 @@ TEST_F(PhysicsSystemFixture, GZ_UTILS_TEST_DISABLED_ON_WIN32(JointsInWorld))
+/// Test ray intersections computed by physics system during Update loop.
+TEST_F(PhysicsSystemFixture, GZ_UTILS_TEST_DISABLED_ON_WIN32(RayIntersections))
+{
+  ServerConfig serverConfig;
+
+  const auto sdfFile =
+    common::joinPaths(PROJECT_SOURCE_PATH, "test", "worlds", "empty.sdf");
+
+  serverConfig.SetSdfFile(sdfFile);
+  Server server(serverConfig);
+  server.SetUpdatePeriod(1ns);
+
+  Entity testEntity1;
+  Entity testEntity2;
+
+  test::Relay testSystem;
+
+  // During PreUpdate, add rays to testEntity1 and testEntity2
+  testSystem.OnPreUpdate(
+      [&](const UpdateInfo &/*_info*/, EntityComponentManager &_ecm)
+      {
+        // Set collision detector to bullet (supports ray intersections).
+        auto worldEntity = _ecm.EntityByComponents(components::World());
+        _ecm.CreateComponent(
+          worldEntity, components::PhysicsCollisionDetector("bullet"));
+
+        // Create MultiRay and MultiRayIntersections components for testEntity1
+        testEntity1 = _ecm.CreateEntity();
+        _ecm.CreateComponent(testEntity1, components::MultiRay());
+        _ecm.CreateComponent(testEntity1, components::MultiRayIntersections());
+
+        // Create MultiRay and MultiRayIntersections components for testEntity2
+        testEntity2 = _ecm.CreateEntity();
+        _ecm.CreateComponent(testEntity2, components::MultiRay());
+        _ecm.CreateComponent(testEntity2, components::MultiRayIntersections());
+
+        // Add 5 rays to testEntity1 that intersect with the ground plane
+        auto &rays1 =
+          _ecm.Component<components::MultiRay>(testEntity1)->Data();
+        for (size_t i = 0; i < 5; ++i)
+        {
+          components::RayInfo ray;
+          ray.start = math::Vector3d(0, 0, 10 - i);
+          ray.end = math::Vector3d(0, 0, -10);
+          rays1.push_back(ray);
+        }
+
+        // Add 2 rays to testEntity2 that don't intersect with the ground plane
+        auto &rays2 = _ecm.Component<components::MultiRay>(testEntity2)->Data();
+        for (size_t i = 0; i < 2; ++i)
+        {
+          components::RayInfo ray;
+          ray.start = math::Vector3d(0, 0, 10 - i);
+          ray.end = math::Vector3d(0, 0, 5);
+          rays2.push_back(ray);
+        }
+      });
+  // Check ray intersections for testEntity1 and testEntity2
+  testSystem.OnPostUpdate(
+      [&](const UpdateInfo &/*_info*/, const EntityComponentManager &_ecm)
+      {
+        // check the raycasting results for testEntity1
+        auto &rays1 = _ecm.Component<components::MultiRay>(testEntity1)->Data();
+        auto &results1 =
+          _ecm.Component<components::MultiRayIntersections>(
+            testEntity1)->Data();
+        ASSERT_EQ(rays1.size(), results1.size());
+
+        for (size_t i = 0; i < results1.size(); ++i) {
+          ASSERT_EQ(results1[i].point, math::Vector3d::Zero);
+          ASSERT_EQ(results1[i].normal, math::Vector3d(0, 0, 1));
+          double exp_fraction =
```

Comment:
nit: use `const`, use `camelCase` style

Reply:
Fixed in https://github.com/gazebosim/gz-sim/pull/2514/commits/34cbfd66e52c57028c8cc35cf19a917e6db41bb2